bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -192,9 +194,13 @@ public void updated(Provider<E> provider, E oldElement, E element) {
                 // the given "oldElement" might not be the live instance but
                 // loaded from storage. operate on the real element:
                 E existingElement = get(oldElement.getUID());
-                beforeUpdateElement(existingElement);
+                if (existingElement != null) {
+                    beforeUpdateElement(existingElement);
+                }
                 onUpdateElement(oldElement, element);
-                elements.remove(existingElement);
+                if (existingElement != null) {
```

Comment:
I do not want to change the order of the method calls. Thus I had to add two `null` checks here. But do we really need this one too? Or is the `Collection.remove()` method capable of handling `null` values properly?

Reply:
It depends on the collection itself.
A collection is allowed to throw a NPE if it doesn't like it...

```
NullPointerException - if the specified element is null and this collection does not permit null elements (optional)
```

We put `CopyOnWriteArraySet` as a collection implementation into the map.
So, we could change the map value type to `CopyOnWriteArraySet` or handle every collection the same way and be save our self here.

Reply:
Okay, I do not want to break the loose coupling. For me it is fine to stay with two `null`checks.